### PR TITLE
reorganize project setup, and maintenance

### PR DESCRIPTION
**Pull request by TacoTheDank**
could you fix the submodule reference (if you merge this) please? i dont know how to
(set it to the latest commit, which is this)
https://github.com/uberspot/2048/commit/7986ea0432fc9fd6634dfeaff3b966ecb7aa78de

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,34 +1,5 @@
-# built application files
```
Comment (uberspot) [inline .gitignore]: Why delete all these? It's healthy to ignore these files.

File: app/src/main/java/com/uberspot/a2048/MainActivity.java [hunk 2]
```diff
@@ -36,12 +32,11 @@
     private long mLastBackPress;
     private static final long mBackPressThreshold = 3500;
     private static final String IS_FULLSCREEN_PREF = "is_fullscreen_pref";
-    private static boolean DEF_FULLSCREEN = true;
```
Comment (uberspot) [inline app/src/main/java/com/uberspot/a2048/MainActivity.java]: Why remove this and hardcode to "true" ? It's a convenient place to change this as an option. Can we reset this?

**Comment by uberspot** [comment]
Looks good in general :) 
I'll gladly merge, but could you first fix the two comments I made and the build failure? 
Here's the Travis CI log in case it helps. :) 
`
Welcome to Gradle 4.10.2!
Here are the highlights of this release:
 - Incremental Java compilation by default
 - Periodic Gradle caches cleanup
 - Gradle Kotlin DSL 1.0-RC6
 - Nested included builds
 - SNAPSHOT plugin versions in the `plugins {}` block
For more details see https://docs.gradle.org/4.10.2/release-notes.html
> Task :app:checkDebugClasspath FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Could not find com.android.support:support-v4:28.0.0.
  Searched in the following locations:
    - file:/usr/local/android-sdk/extras/m2repository/com/android/support/support-v4/28.0.0/support-v4-28.0.0.pom
    - file:/usr/local/android-sdk/extras/m2repository/com/android/support/support-v4/28.0.0/support-v4-28.0.0.jar
    - file:/usr/local/android-sdk/extras/google/m2repository/com/android/support/support-v4/28.0.0/support-v4-28.0.0.pom
    - file:/usr/local/android-sdk/extras/google/m2repository/com/android/support/support-v4/28.0.0/support-v4-28.0.0.jar
    - file:/usr/local/android-sdk/extras/android/m2repository/com/android/support/support-v4/28.0.0/support-v4-28.0.0.pom
    - file:/usr/local/android-sdk/extras/android/m2repository/com/android/support/support-v4/28.0.0/support-v4-28.0.0.jar
  Required by:
      project :app
> Could not find de.cketti.library.changelog:ckchangelog:1.2.2.
  Searched in the following locations:
    - file:/usr/local/android-sdk/extras/m2repository/de/cketti/library/changelog/ckchangelog/1.2.2/ckchangelog-1.2.2.pom
    - file:/usr/local/android-sdk/extras/m2repository/de/cketti/library/changelog/ckchangelog/1.2.2/ckchangelog-1.2.2.jar
    - file:/usr/local/android-sdk/extras/google/m2repository/de/cketti/library/changelog/ckchangelog/1.2.2/ckchangelog-1.2.2.pom
    - file:/usr/local/android-sdk/extras/google/m2repository/de/cketti/library/changelog/ckchangelog/1.2.2/ckchangelog-1.2.2.jar
    - file:/usr/local/android-sdk/extras/android/m2repository/de/cketti/library/changelog/ckchangelog/1.2.2/ckchangelog-1.2.2.pom
    - file:/usr/local/android-sdk/extras/android/m2repository/de/cketti/library/changelog/ckchangelog/1.2.2/ckchangelog-1.2.2.jar
  Required by:
      project :app
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 31s
1 actionable task: 1 executed
The command "./gradlew build" exited with 1.
Done. Your build exited with 1.`

Also you can update the submodules to point to the latest commits with this:
`git submodule foreach git pull origin master` from https://stackoverflow.com/questions/5828324/update-git-submodule-to-latest-commit-on-origin/21195182 
Hope it helps :)

File: .gitignore (hunk 1 above)
Comment (TacoTheDank) [reply .gitignore]: its already covered by the root .gitignore file, although if you'd like to keep it that's fine

**Comment by TacoTheDank** [comment]
doing redo of entire thing, to see why it travis wasnt building, also just fixing sucky code

**Comment by TacoTheDank** [comment]
this is MUCH better tbh

**Comment by uberspot** [comment]
Merged :) nice work

**Pull request merged**